Repository: plone.dexterity


Branch: refs/heads/master
Date: 2020-09-24T00:27:59+02:00
Author: Maurits van Rees (mauritsvanrees) <maurits@vanrees.org>
Commit: https://github.com/plone/plone.dexterity/commit/69cbb406379e5cafd8bc629516ce2551cbfe5622

Fixed missing ISiteRoot utility when running tests with Zope 5.

You get this error once:

```
Error in test test_form_create (plone.dexterity.tests.test_views.TestAddView)
Traceback (most recent call last):
  File "/Users/maurits/.pyenv/versions/3.8.5/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/Users/maurits/.pyenv/versions/3.8.5/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/Users/maurits/.pyenv/versions/3.8.5/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/Users/maurits/community/plone-coredev/6.0/src/plone.dexterity/plone/dexterity/tests/test_views.py", line 129, in test_form_create
    self.assertEqual(obj_dummy, form.create(data_dummy))
  File "/Users/maurits/community/plone-coredev/6.0/src/plone.dexterity/plone/dexterity/browser/add.py", line 79, in create
    if IAcquirer.providedBy(content):
  File "/Users/maurits/community/plone-coredev/6.0/src/plone.dexterity/plone/dexterity/content.py", line 172, in __get__
    main_schema = SCHEMA_CACHE.get(portal_type)
  File "/Users/maurits/community/plone-coredev/6.0/src/plone.dexterity/plone/dexterity/schema.py", line 110, in decorator
    value = func(self, fti)
  File "/Users/maurits/community/plone-coredev/6.0/src/plone.dexterity/plone/dexterity/schema.py", line 157, in get
    return fti.lookupSchema()
  File "/Users/maurits/community/plone-coredev/6.0/src/plone.dexterity/plone/dexterity/fti.py", line 268, in lookupSchema
    schemaName = portalTypeToSchemaName(self.getId())
  File "/Users/maurits/community/plone-coredev/6.0/src/plone.dexterity/plone/dexterity/schema.py", line 328, in portalTypeToSchemaName
    prefix = '/'.join(getUtility(ISiteRoot).getPhysicalPath())[1:]
  File "/Users/maurits/shared-eggs/cp38/zope.component-4.6.2-py3.8.egg/zope/component/_api.py", line 165, in getUtility
    raise ComponentLookupError(interface, name)
zope.interface.interfaces.ComponentLookupError: (&lt;InterfaceClass Products.CMFCore.interfaces.ISiteRoot&gt;, '')
```

Note that this is in a true unittest without layers, so without any Plone Site being setup.
So the test failure seems innocent.
The fix is to mock the site root in the same way as we already do in lots of other tests.

I don't know why this fails with Zope 5.
It could be because Zope 5 recently upgraded zope.interface to 5.1.0.

Files changed:
A news/680.bugfix
M plone/dexterity/tests/test_views.py

b'diff --git a/news/680.bugfix b/news/680.bugfix\nnew file mode 100644\nindex 0000000..cd7bb0d\n--- /dev/null\n+++ b/news/680.bugfix\n@@ -0,0 +1,2 @@\n+Fixed missing ISiteRoot utility when running tests with Zope 5.\n+[maurits]\ndiff --git a/plone/dexterity/tests/test_views.py b/plone/dexterity/tests/test_views.py\nindex 852c437..bb7e550 100644\n--- a/plone/dexterity/tests/test_views.py\n+++ b/plone/dexterity/tests/test_views.py\n@@ -20,6 +20,7 @@\n from plone.dexterity.interfaces import IEditFinishedEvent\n from plone.dexterity.schema import SCHEMA_CACHE\n from plone.z3cform.interfaces import IDeferSecurityCheck\n+from Products.CMFCore.interfaces import ISiteRoot\n from Products.statusmessages.interfaces import IStatusMessage\n from z3c.form.action import Actions\n from z3c.form.datamanager import AttributeField\n@@ -126,6 +127,9 @@ class ISchema(Interface):\n \n         provideAdapter(AttributeField)\n \n+        portal = self.create_dummy(getPhysicalPath=lambda: (\'\', \'site\'))\n+        self.mock_utility(portal, ISiteRoot)\n+\n         self.assertEqual(obj_dummy, form.create(data_dummy))\n         self.assertEqual("testtype", obj_dummy.portal_type)\n \n'

Repository: plone.dexterity


Branch: refs/heads/master
Date: 2020-09-24T09:52:11+02:00
Author: Jens W. Klein (jensens) <jk@kleinundpartner.at>
Commit: https://github.com/plone/plone.dexterity/commit/5b05b15d5960142ffa8d85239008648bf345668b

Merge pull request #134 from plone/zope5

Fixed missing ISiteRoot utility when running tests with Zope 5.

Files changed:
A news/680.bugfix
M plone/dexterity/tests/test_views.py

b'diff --git a/news/680.bugfix b/news/680.bugfix\nnew file mode 100644\nindex 0000000..cd7bb0d\n--- /dev/null\n+++ b/news/680.bugfix\n@@ -0,0 +1,2 @@\n+Fixed missing ISiteRoot utility when running tests with Zope 5.\n+[maurits]\ndiff --git a/plone/dexterity/tests/test_views.py b/plone/dexterity/tests/test_views.py\nindex 852c437..bb7e550 100644\n--- a/plone/dexterity/tests/test_views.py\n+++ b/plone/dexterity/tests/test_views.py\n@@ -20,6 +20,7 @@\n from plone.dexterity.interfaces import IEditFinishedEvent\n from plone.dexterity.schema import SCHEMA_CACHE\n from plone.z3cform.interfaces import IDeferSecurityCheck\n+from Products.CMFCore.interfaces import ISiteRoot\n from Products.statusmessages.interfaces import IStatusMessage\n from z3c.form.action import Actions\n from z3c.form.datamanager import AttributeField\n@@ -126,6 +127,9 @@ class ISchema(Interface):\n \n         provideAdapter(AttributeField)\n \n+        portal = self.create_dummy(getPhysicalPath=lambda: (\'\', \'site\'))\n+        self.mock_utility(portal, ISiteRoot)\n+\n         self.assertEqual(obj_dummy, form.create(data_dummy))\n         self.assertEqual("testtype", obj_dummy.portal_type)\n \n'

